modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,758 @@
+            for item,amount in pairs(items) do
+                player.insert {name = item, count = amount}
+            end
+        end
+    else
+        player.set_controller {type = defines.controllers.spectator}
+        player.teleport(variables.level.center,variables.level.surface)
+    end
+
+
+end
+
+--gui
+local speed_slider
+local  function speed_change(player, _, _)
+    local Main_gui = Gui.get_left_element(player, game_gui)
+    local table = Main_gui.container["slider"].table
+    game.speed = table[speed_slider.name].slider_value
+end
+
+--game gui
+
+speed_slider =
+Gui.element{
+    type = "slider",
+    minimum_value = 1,
+    maximum_value  = 64,
+    value = 4,
+    value_step = 1,
+}:on_value_changed(speed_change)
+local label_func =
+Gui.element(function(_,parent,name,style,caption)
+    if name ~= nil then
+        return parent.add {
+            type = "label",
+            caption = caption,
+            style = style,
+            name = name
+        }
+    else
+        return parent.add {
+            type = "label",
+            caption = caption,
+            style = style,
+        }
+    end
+end)
+game_gui =
+Gui.element(function(event_trigger,parent)
+    local container = Gui.container(parent,event_trigger,200)
+    Gui.header(container,"Tight spot","The Tight money manual.",true)
+
+    local scroll_table_slider = Gui.scroll_table(container,250,2,"slider")
+    label_func(scroll_table_slider,nil,"heading_1_label","speed:")
+    speed_slider(scroll_table_slider)
+    scroll_table_slider.visible  = false
+
+    local tilel1 = label_func(container,nil,"heading_1_label","Money:")
+    tilel1.style.left_padding = 7
+
+    local scroll_table_labels = Gui.scroll_table(container,250,2,"Money")
+    local scroll_table_labels_style = scroll_table_labels.style
+    scroll_table_labels_style.top_cell_padding = 3
+    scroll_table_labels_style.bottom_cell_padding = 3
+    scroll_table_labels_style.left_cell_padding  = 7
+
+    local tilel2 = label_func(container,nil,"heading_1_label","Objective:")
+    tilel2.style.left_padding = 7
+
+    local scroll_table2 = Gui.scroll_table(container,250,2,"Objectives")
+    local scroll_table2_style = scroll_table2.style
+    scroll_table2_style.top_cell_padding = 3
+    scroll_table2_style.bottom_cell_padding = 3
+    scroll_table2_style.left_cell_padding  = 7
+
+    label_func(scroll_table_labels,nil,"label","Balance: ")
+    label_func(scroll_table_labels,"Balance","label","0")
+
+    label_func(scroll_table_labels,nil,"label","Debt: ")
+    label_func(scroll_table_labels,"Debt","label","0")
+
+    label_func(scroll_table_labels,nil,"label","Time: ")
+    label_func(scroll_table_labels,"Time","label","10:00")
+
+    label_func(scroll_table2,nil,"label","Type:")
+    label_func(scroll_table2,"Objective","label","Iron gear")
+
+    label_func(scroll_table2,nil,"label","Price: ")
+    label_func(scroll_table2,"prices","label","15")
```

Comment:
Firstly, I would remove the use of label_func from the two header labels.
This would mean you no longer to to include the style any more.
It would aklso mean you could change the function to make two labels each call:
`label_func(parent, name, caption, default)` -> `label_func(scroll_table2, 'prices', 'Price: ', '15')`

Reply:
ill think about it 👍